typescript/packages/plugins/balancer/src/balancer.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Tool } from "@goat-sdk/core";
+import type { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { BalancerApi, ChainId, Slippage, SwapKind, Token, Swap, AddLiquidity, AddLiquidityKind, TokenAmount } from "@balancer/sdk";
+import { SwapParameters, LiquidityParameters } from "./parameters";
+
+export class BalancerService {
+    @Tool({
+        description: "Performs a token swap on Balancer using their Smart Order Router",
+    })
+    async swap(walletClient: EVMWalletClient, parameters: SwapParameters) {
```

Comment:
You might want to add a name if the function name is just swap. The name needs to be descriptive enough for the LLM (e.g `swapOnBalancer` or adding it to the @Tool decorator, e.g `name: "swap_on_balancer"`)

Reply:
done , sounds good :)